internal/controller/apiextensions/composition/reconciler.go

```diff
@@ -180,6 +176,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 			continue
 		}
 
+		// Make sure this revision is owned/controlled by the Composition.
+		// Note(turkenh): Owner references are stripped out when a resource is
+		// moved from one cluster to another (i.e. backup/restore) since the UID
+		// of the owner is not preserved. We need to make sure to re-add the
+		// owner reference here.
+		meta.AddOwnerReference(rev, meta.AsController(meta.TypedReferenceTo(comp, v1.CompositionGroupVersionKind)))
+
```

Comment:
I think we should move this logic before the following check 
```
if rev.GetLabels()[v1.LabelCompositionHash] != currentHash[:63] {
	continue
}
```
Otherwise, we cannot set the owner reference to inactive revisions.

Reply:
https://pkg.go.dev/github.com/crossplane/crossplane-runtime@v1.14.3/pkg/meta#AddControllerReference

Would it make sense to use AddControllerReference rather than AddOwnerReference?

I think AddOwnerReference will make this composition the controller of the revision, even if another composition (somehow) already controls it. AddControllerReference will make the composition the controller only if it has no controller reference. If the revision already has a controller it will return an error.